tensorflow_addons/image/distort_image_ops.py

```diff
@@ -99,6 +102,43 @@ def random_hsv_in_yiq(
         )
 
 
+def _adjust_hsv_in_yiq(
+    image,
+    delta_hue,
+    scale_saturation,
+    scale_value,
+):
+    if image.shape.rank is not None and image.shape.rank < 3:
+        raise ValueError("input must be at least 3-D.")
+    if image.shape[-1] is not None and image.shape[-1] != 3:
+        raise ValueError(
+            "input must have 3 channels but instead has {}.".format(image.shape[-1])
+        )
+    # Construct hsv linear transformation matrix in YIQ space.
+    # https://beesbuzz.biz/code/hsv_color_transforms.php
+    yiq = tf.constant(
+        [[0.299, 0.596, 0.211], [0.587, -0.274, -0.523], [0.114, -0.322, 0.312]],
+        dtype=tf.float32,
+    )
+    yiq_inverse = tf.constant(
+        [
+            [1.0, 1.0, 1.0],
+            [0.95617069, -0.2726886, -1.103744],
+            [0.62143257, -0.64681324, 1.70062309],
+        ],
+        dtype=tf.float32,
+    )
+    vsu = scale_value * scale_saturation * tf.math.cos(delta_hue)
+    vsw = scale_value * scale_saturation * tf.math.sin(delta_hue)
+    hsv_transform = tf.convert_to_tensor(
+        [[scale_value, 0, 0], [0, vsu, vsw], [0, -vsw, vsu]], dtype=tf.float32
+    )
+    transform_matrix = yiq @ hsv_transform @ yiq_inverse
+
+    image = image @ transform_matrix
```

Comment:
Do we need to support other `dtypes` as in https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/ops/image_ops_impl.py#L2522-L2525 ?

Reply:
Thanks for reminding! That's reasonable, maybe let me do it in the next PR?